scripts/commit-msg

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env ruby
+
+# This script is intended to be used under repo_dir/.git/hooks to check
+# each git commit message against OMR contributing guidelines. The script
+# should be executable and it will be invoked when a 'git commit' command 
+# is issued.
+
+message_file = ARGV[0]
+message = File.read(message_file)
+
+# remove git generated messages
+commit = ""
+message.each_line do |line|
+  if (line.match(/^(?!#).*$/))
+    commit.concat(line)
+  end
+end
+
+title, body = commit.split("\n", 2)
+
+# check commit title length
+title_len = title.length
+#puts "Title is #{title}, length is #{title_len}"
+
+if (title_len > 0 && title_len < 50)
+  if (title.match(/^[A-Z]/) && title.match(/[a-zA-Z0-9]$/))
+    state = "success"
+    description = "Commit title satisfies commit guidelines."
+  else
+    state = "failure"
+    description = "Commit title should begin with a capital letter and" \
+                  " does not end in a punctuation."
+  end
+else
+  state = "failure"
+  description = "Commit title should be less than 50 characters: actual #{title_len}"
+end
+
+# check commit message body length
+if (state == "success")
+  body.each_line.with_index() do |line, j|
+    line.chomp!
+#    puts "Line length is #{line.length}"
+    if (line.length >= 0 && line.length <= 72)
```

Comment:
The 72 character limit is really only a recommendation in the CG.  There are circumstances where it may be difficult or inconvenient to ensure a 72 char limit (e.g., pasting source code, or an entry from a compiler log).  Hence, exceeding 72 chars should illicit an informational warning that some lines have exceeded 72 chars and the message body should be checked manually.